src/mmw/js/src/core/models.js

```diff
@@ -238,6 +239,77 @@ var LayerModel = Backbone.Model.extend({
     }
 });
 
+var WeatherStationLayerModel = LayerModel.extend({
+    defaults: {
+        bringToFront: true, // Ensures popups work when station is within project AOI.
+        activeWeatherStations: null,
+        activeWeatherStationPoints: null
+    },
+
+    // Sets active weather stations. Used to change the visual appearance
+    // of any stations participating as modeling input, and to set the
+    // distance of the weather station from an AOI into the popup.
+    setActiveWeatherStations: function(weatherStations) {
+        var leafletLayer = this.get('leafletLayer'),
+            activeWeatherStations = {};
+
+        if(this.activeWeatherStations) {
+            this.clearActiveWeatherStations();
+        }
+
+        _.forEach(weatherStations, function(ws) {
+            activeWeatherStations[ws.station] = ws;
+        });
+
+
+        this.set('activeWeatherStations', activeWeatherStations);
+
+        if(leafletLayer) {
+            var activeWeatherStationPoints =
+                weatherStationLayer.Layer.setActiveWeatherStations(leafletLayer, activeWeatherStations);
+            this.set('activeWeatherStationPoints', activeWeatherStationPoints);
+        }
+    },
+
+    clearActiveWeatherStations: function() {
+        var leafletLayer = this.get('leafletLayer'),
+            prevActiveWeatherStations = this.get('activeWeatherStations');
+
+        this.set('activeWeatherStations', null);
+        this.set('activeWeatherStationPoints', null);
+
+        if(leafletLayer) {
+            weatherStationLayer.Layer.clearActiveWeatherStations(leafletLayer, prevActiveWeatherStations);
+        }
+    },
+
+    // Toggles the layer. If toggling on, zoom the map to fit both
+    // the AOI and the active weather stations.
+    toggleAndZoom: function(areaOfInterest, map, layerGroup) {
+        utils.toggleLayer(this, map, layerGroup);
+
+        if(this.get('active')) {
+            // We toggled on; zoom to the active weather stations.
+            var activeStationPoints = this.get('activeWeatherStationPoints'),
```

Comment:
Moved this code to use the model `get` and `set`, which I wasn't doing beforehand mistakenly.